### ISSUE TITLE
OOM Issues despite plenty of resources on M1 Mac

### ISSUE THREAD

**Issue by david-engelmann** [pytorch/pytorch#79283]
### 🐛 Describe the bug

In a few different instances, I've run into an issue where during the loss.backward() suddenly the program will return "Killed". The first few iterations loss.backward() will be successful and then it just dies. I've tracked the disk usage as part of my troubleshooting and tons of memory remain before it fails. It only appears to be happening on my M1 Mac Mini. If I shrink the dimensions immensely such as making image_size = 100 or vocab_size = 1000, it's able to finish running (through with obviously degraded resutls). I've tried strategically calling gc.collect() with no success. 

I wish I had more helpful information, but as far as I can tell the problem is two fold. First part is that the M1 Mac GPU is not recognized ie. torch.cuda.is_available() returns false and "mps" isn't recognized as an available device type in torch version 1.11.0. This means "cpu" is the only working device option. Second part seems to be the freeing of the cache.

### Versions

Python Version: 3.9
Ubuntu Version: 21.04 (dev-hirsute)
PyTorch Version: 1.11.0
TorchVision Version: 0.12.0

**Comment by kulinseth** [comment]
Hi @david-engelmann , thanks for the issue.
`First part is that the M1 Mac GPU is not recognized ie. torch.cuda.is_available() returns false and "mps" `
https://pytorch.org/docs/master/notes/mps.html. For Mac GPU acceleration we have added `mps` device. And to use it:

```
# Check that MPS is available
if not torch.backends.mps.is_available():
    if not torch.backends.mps.is_built():
        print("MPS not available because the current PyTorch install was not "
              "built with MPS enabled.")
    else:
        print("MPS not available because the current MacOS version is not 12.3+ "
              "and/or you do not have an MPS-enabled device on this machine.")

else:
    mps_device = torch.device("mps")
```

The MPS device will be available in 1.12 stable release onwards. Currently you can use the Nightly binaries in the selector to get the support.

<img width="1043" alt="Screen Shot 2022-06-14 at 4 51 38 PM" src="https://user-images.githubusercontent.com/453004/173708001-7bc84f6d-dc00-4e98-a996-182cc8480853.png">


`I've tracked the disk usage as part of my troubleshooting and tons of memory remain before it fails. It only appears to be happening on my M1 Mac Mini. If I shrink the dimensions immensely such as making image_size = 100 or vocab_size = 1000, it's able to finish running (through with obviously degraded resutls). I've tried strategically calling gc.collect() with no success.`

Can you please provide a reproducible case or a link to the network you are training?

**Comment by david-engelmann** [comment]
@kulinseth Amazing for point one, for now I can use nightly and it will be in the 1.12 release. For the second point, I'll set up a repo with a minimum viable example and comment it here!

**Comment by razarmehr** [comment]
Hi @david-engelmann . I tested with the sample provided and the memory seems to be stable now. Could you please test with the latest nightly build and check if you see the issue? Thank you.

**Comment by david-engelmann** [comment]
@razarmehr I was able to run successfully with the nightly build by adding more resources to my docker desktop. Hopefully, the recent updates help as well.
